### DISCUSSION TITLE
Code re-use

### DISCUSSION THREAD

**Discussion by ghammad** [activityMonitoring/biobankAccelerometerAnalysis#266] [General]
Hello biobank people,

I am the main developper of the [pyActigraphy](https://github.com/ghammad/pyActigraphy) package, an analysis package for actigraphy data. It provides an interface to the file format the `accelerometer` package produces with `AccProcess` (XXX-timeSeries.csv.gz). As such, upon reading this file, I needed to parse the date time columns and I wanted to let the users decide whether or not to impute the missing data. Therefore, I import your package and use your functions ([date_parser](https://github.com/OxWearables/biobankAccelerometerAnalysis/blob/ed0caad712a9686deb2bf3a37273c1dedd14beb6/src/accelerometer/utils.py#L342) and [imputMissing](https://github.com/OxWearables/biobankAccelerometerAnalysis/blob/ed0caad712a9686deb2bf3a37273c1dedd14beb6/src/accelerometer/summarisation.py#L218)).

However, adding `accelerometer` as a dependency has created installation issues for users with recent python installations, as your requirement `numpy=1.21.*` is incompatible with recent numba version, which I do require in pyActigraphy dependencies.

Following up on that issue, I received a [pull request](https://github.com/ghammad/pyActigraphy/pull/160#pullrequestreview-2501933845) from a user that would like to re-implement these two functions within the `pyActigraphy` package. 

While I initially thought that importing `accelerometer` was a straightforward way to link our packages (I like the idea of an ecosystem of packages for actigraphy data), it appears, as a user feedback, that it is most likely an overkill, given that it uses only two rather generic functions but adds complexity from the dependency point of view, up to the point where it sometimes prevent users from installing pyActigraphy.

I thus propose, as these functions are rather generic and there are not multiple ways to implement them (especially the date_parser function, 4 lines of codes), to add within the function description in pyActigraphy an explicit reference to the `accelerometer` package , its version and its license. 

I hope that you will find this solution acceptable.

I'll wait for your feedback before accepting the PR.

Best regards,

Grégory

**Comment by aidendoherty** [comment]
Thank you @ghammad
@chanshing and I will discuss and aim to get back to you this week

**Comment by aidendoherty** [comment]
Dear @ghammad 
We like your suggestion to add within the function description of date_parser and imputeMissing in pyActigraphy an explicit reference to the accelerometer package, its version and its license. In particular, it would be good to note that this licence allows for free academic use, but that permission would need to be sought from the University of Oxford for commercial use.

Is this ok with you?

Best wishes
Aiden

**Comment by ghammad** [comment]
Dear @aidendoherty , @chanshing 

Thank you for the prompt reply and positive outcome. I will thus add an explicit mention about the potential commercial use (after permission from the University of Oxford).

You can follow the development and resolution of the PR [here](https://github.com/ghammad/pyActigraphy/pull/160).

Best regards,

Grégory

**Reply by aidendoherty** [reply]
Sounds great!
